### PR TITLE
feat(pg): Install Sphinx fork during sphinx install

### PR DESCRIPTION
**Pull request by RPate97**
## Purpose
- Updates the `sphinx install` command to install the Sphinx Foundy fork binaries
- Updates installation docs to no longer recommend installing the fork via `foundryup`
- Updates ci guide to recommend running `yarn sphinx install —confirm` instead of using the foundry GitHub action integration

## Todo 
- [ ] Test CI guide

### CODE REVIEW

File: docs/cli-existing-project.md [hunk 1]
```diff
@@ -13,24 +13,23 @@ In this guide, you'll propose the deployment on the command line and then approv
...
 
 1. [Prerequisites](#1-prerequisites)
-2. [Install Sphinx Foundry Fork](#2-install-sphinx-foundry-fork)
-3. [Install Sphinx CLI](#3-install-sphinx-cli)
-4. [Install Sphinx Foundry library](#4-install-sphinx-foundry-library)
-5. [Update `.gitignore`](#5-update-gitignore)
-6. [Add remapping](#6-add-remapping)
-7. [Update your deployment script](#7-update-your-deployment-script)\
+2. [Install Sphinx CLI](#2-install-sphinx-cli)
+3. [Install Sphinx Foundry library](#3-install-sphinx-foundry-library)
```
Comment (sam-goldman) [inline docs/cli-existing-project.md:17]: #3 needs to be updated (the name and the link, which is dead)

File: docs/cli-existing-project.md [hunk 2]
```diff
@@ -44,22 +43,7 @@ In this guide, you'll propose the deployment on the command line and then approv
   * [Node Version >=16.16.0](https://nodejs.org/en/download). (Run `node -v` to see your current version).
   * [Rust Compiler](https://www.rust-lang.org/tools/install)
 
-## 2. Install Sphinx Foundry Fork
-For now, Sphinx requires a [fork of Foundry](https://github.com/sphinx-labs/foundry) that resolves several bugs that conflict with Sphinx. We're working on getting these fixed in the official Foundry repo.
-
-If you are using macOS, you will first need to install the Xcode Command Line Tools:
-```
-xcode-select --install
```
Comment (sam-goldman) [inline docs/cli-existing-project.md:52]: Is installing xcode not necessary anymore?

File: docs/cli-existing-project.md [hunk 3]
```diff
@@ -44,22 +43,7 @@ In this guide, you'll propose the deployment on the command line and then approv
...
-```
-
-Then, install the Sphinx Foundry fork with the following command. The installation process may take several minutes to complete.
-```
-foundryup --repo sphinx-labs/foundry --branch sphinx-patch-v0.1.0
-```
-
-> If you run into problems installing our Foundry fork, [try steps in the troubleshooting guide](https://github.com/sphinx-labs/sphinx/blob/main/docs/troubleshooting-guide.md#installing-sphinxs-foundry-fork). If you are still unable to resolve the issue, please reach out in the [Discord](https://discord.gg/7Gc3DK33Np). We're always happy to help.
```
Comment (sam-goldman) [inline docs/cli-existing-project.md:60]: > If you run into problems installing our Foundry fork, [try steps in the troubleshooting guide](https://github.com/sphinx-labs/sphinx/blob/main/docs/troubleshooting-guide.md#installing-sphinxs-foundry-fork). If you are still unable to resolve the issue, please reach out in the [Discord](https://discord.gg/7Gc3DK33Np). We're always happy to help.

Should we keep this? Looks like it's removed in this PR

Comment (sam-goldman) [reply docs/cli-existing-project.md:60]: If you don't think the troubleshooting guide is necessary anymore, can you delete it?

File: docs/cli-existing-project.md (hunk 2 above)
Comment (RPate97) [reply docs/cli-existing-project.md:52]: No it should not be since this installation method does not require the compiler.

Comment (sam-goldman) [reply docs/cli-existing-project.md:52]: Sweet

File: docs/cli-existing-project.md [hunk 4]
```diff
@@ -80,9 +64,12 @@ pnpm:
 pnpm add -D @sphinx-labs/plugins
 ```
 
-## 4. Install Sphinx Foundry Library
 
-Use the `sphinx install` command to install the Sphinx Foundry library.
+## 3. Install Sphinx Foundry Library and Fork
+
+Use the `sphinx install` command to install the Sphinx Foundry library and Foundry fork.
```
Comment (sam-goldman) [inline docs/cli-existing-project.md:70]: Love that this is in a single command

File: docs/cli-quickstart.md [hunk 5]
```diff
@@ -62,25 +61,12 @@ pnpm:
...
-```
-xcode-select --install
-```
-
-Then, install the Sphinx Foundry fork with the following command. The installation process may take several minutes to complete.
-```
-foundryup --repo sphinx-labs/foundry --branch sphinx-patch-v0.1.0
-```
-
-> If you run into problems installing our Foundry fork, [try steps in the troubleshooting guide](https://github.com/sphinx-labs/sphinx/blob/main/docs/troubleshooting-guide.md#installing-sphinxs-foundry-fork). If you are still unable to resolve the issue, please reach out in the [Discord](https://discord.gg/7Gc3DK33Np). We're always happy to help.
```
Comment (sam-goldman) [inline docs/cli-quickstart.md:78]: > If you run into problems installing our Foundry fork, [try steps in the troubleshooting guide](https://github.com/sphinx-labs/sphinx/blob/main/docs/troubleshooting-guide.md#installing-sphinxs-foundry-fork). If you are still unable to resolve the issue, please reach out in the [Discord](https://discord.gg/7Gc3DK33Np). We're always happy to help.

Should we keep this? Looks like it's removed in this PR

File: packages/plugins/src/cli/install/index.ts [hunk 6]
```diff
@@ -1,10 +1,29 @@
...
+import { sphinxUp } from './sphinxup'
+
+const installWithUpdate = async (
+  spinner: ora.Ora,
+  skipLibrary: boolean,
+  confirm: boolean
+) => {
+  if (!confirm) {
+    await userConfirmation(
+      'Would you like to install the Sphinx Foundry fork? This is required to use Sphinx and will replace your existing foundry instllation.\nConfirm? (y\\n):'
```
Comment (sam-goldman) [inline packages/plugins/src/cli/install/index.ts:16]: foundry -> Foundry

Comment (sam-goldman) [inline packages/plugins/src/cli/install/index.ts:16]: instllation -> installation

File: packages/plugins/src/cli/setup.ts [hunk 7]
```diff
@@ -136,10 +136,25 @@ export const makeCLI = (
...
+        y
+          .usage('Usage: sphinx install')
+          .option('skip-library', {
+            describe:
+              'Skips installing the Sphinx library and only installs the Sphinx Foundry fork.',
+            boolean: true,
+            default: false,
+          })
+          .option(confirmOption, {
+            describe: 'Confirm the deployment without displaying a preview.',
```
Comment (sam-goldman) [inline packages/plugins/src/cli/setup.ts:149]: This `describe` sentence doesn't make sense in this context

File: packages/plugins/src/cli/setup.ts [hunk 8]
```diff
@@ -136,10 +136,25 @@ export const makeCLI = (
     .command(
       'install',
       'Installs the required version of the Sphinx Solidity library contracts',
```
Comment (sam-goldman) [inline packages/plugins/src/cli/setup.ts:138]: Could update this description to mention the Foundry fork

**Review by sam-goldman — APPROVED** [review]
Looks great! Just some minor comments

**Comment by sam-goldman** [comment]
If we're keeping the troubleshooting section, we need to update it to use `sphinx install` instead of `foundryup`

File: docs/cli-existing-project.md (hunk 3 above)
Comment (RPate97) [reply docs/cli-existing-project.md:60]: I don’t think the troubleshooting guide is necessary. I’ll remove it. We can reintroduce it if we see people running into problems with the installation process.

**Pull request merged**